src/db/mod.rs

```diff
@@ -0,0 +1,209 @@
+use crate::types::BlockHeaderWithFullTransaction;
+use anyhow::{Context, Result};
+use chrono::{TimeZone, Utc};
+use log::{info, warn};
+use sqlx::QueryBuilder;
+use sqlx::{postgres::PgPoolOptions, Pool, Postgres};
+use std::sync::Arc;
+use tokio::sync::OnceCell;
+
+static DB_POOL: OnceCell<Arc<Pool<Postgres>>> = OnceCell::const_new();
+
+pub async fn get_db_pool() -> Result<Arc<Pool<Postgres>>> {
+    match DB_POOL.get() {
+        Some(pool) => Ok(pool.clone()),
+        None => {
+            let pool = PgPoolOptions::new()
+                .max_connections(5)
```

Comment:
```suggestion
                .max_connections(100)
```
Hi Alex, would it be possible to allow for 100 max connections for this?
For the `Update` mode it runs up to 100 threads at once, so allowing more connections would help speed up the writes